File: Retos/Reto #12 - VIERNES 13 [Fácil]/javascript/test0n3.js
const friday13 = (date) => {
  const [year, month] = date;
  return new Date(year, month - 1, 13).getDay() == 5;
};

const tests = {
  input: [
    [2023, 3],
    [2023, 10],
    [2024, 9],
    [2023, 2],
  ],
  output: [false, true, true, false],
};

let errors = 0;

tests["input"].forEach((test, index) => {
  const resp = friday13(test);
  const expected = tests["output"][index];
  if (resp == expected) return;

  errors += 1;
  console.log(`\n\noriginal: ${test}`);
  console.log(resp);
  console.log(`expected: ${expected}`);
});

console.log(`\nTests${errors != 0 ? " not " : " "}passed, ${errors} errors\n`);

File: Retos/Reto #12 - VIERNES 13 [Fácil]/ruby/test0n3.rb
# frozen_string_literal: true

require 'date'

def friday13?(date)
  Date.new(*date, 13).friday?
end

TESTS = { 'input' => [[2023, 3], [2023, 10], [2024, 9], [2023, 2]],
          'output' => [false, true, true, false] }.freeze

errors = 0

TESTS['input'].each_with_index do |test, index|
  resp = friday13?(test)
  expected = TESTS['output'][index]
  next if resp == expected

  errors += 1
  puts "\n\noriginal: #{test}"
  puts resp
  puts "expected: #{expected}"
end

puts "\nTests#{errors != 0 ? ' not ' : ' '}passed, #{errors} errors\n"
